lib/SPIRV/SPIRVWriter.cpp

```diff
@@ -626,17 +626,74 @@ SPIRVType *LLVMToSPIRVBase::transSPIRVJointMatrixINTELType(
   else
     llvm_unreachable("Unexpected type for matrix!");
 
-  auto ParseInteger = [this](StringRef Postfix) -> ConstantInt * {
+  auto ParseInteger = [](StringRef Postfix) -> uint32_t {
     unsigned long long N = 0;
     consumeUnsignedInteger(Postfix, 10, N);
-    return getUInt32(M, N);
+    return N;
   };
-  std::vector<SPIRVValue *> Args;
+  std::vector<uint32_t> ProcessedPostfixes;
   for (size_t I = 1; I != Postfixes.size(); ++I)
-    Args.emplace_back(transConstant(ParseInteger(Postfixes[I])));
+    ProcessedPostfixes.emplace_back(ParseInteger(Postfixes[I]));
```

Comment:
So, here we are parsing the  "10_2_0_0" of 
%spirv.JointMatrixINTEL._bfloat16_10_2_0_0, right? instead of parsing each dimension of arrays?

Reply:
The flow was the following:
We have a multidimensional array to process by RegularizeLLVM pass (it's W/A). It was changing a matrix name to have a postfix like you've mentioned.
Then SPIRVWriter looks at this postfix and parse it to obtain rows, cols etc.
So this patch doesn't change this flow.

Note, that in a close future we will remove this nasty array processing, since now DPCPP FE can create an opaque matrix structure with the correct name (see https://github.com/intel/llvm/pull/6535 )